src/main/java/org/opensearch/timeseries/rest/handler/AbstractTimeSeriesActionHandler.java

```diff
@@ -663,28 +616,104 @@ private void onGetConfigResponse(GetResponse response, boolean indexingDryRun, S
+                    // however, since enabling the flattened result index creates additional resources and due to bwc concerns,
+                    // we do not allow customers to enable this feature for existing running detectors.
+                    listener
+                        .onFailure(
+                            new OpenSearchStatusException(CommonMessages.CAN_NOT_CHANGE_FLATTEN_RESULT_INDEX, RestStatus.BAD_REQUEST)
+                        );
+                    return;
+                }
             } else {
                 if (!ParseUtils.listEqualsWithoutConsideringOrder(existingConfig.getCategoryFields(), config.getCategoryFields())
                     || !Objects.equals(existingConfig.getCustomResultIndexOrAlias(), config.getCustomResultIndexOrAlias())) {
                     breakingUIChange = true;
                 }
             }
-            handleFlattenResultIndexMappingUpdate(existingConfig, listener);
+            ActionListener<Void> confirmBatchRunningListener;
 
-            ActionListener<Void> confirmBatchRunningListener = ActionListener
-                .wrap(
-                    r -> searchConfigInputIndices(id, indexingDryRun, listener),
-                    // can't update config if there is task running
-                    listener::onFailure
-                );
+            // when unselecting flatten result index, need to clean up alias and ingest pipeline resources
+            if (existingConfig.getFlattenResultIndexMapping()
+                && !config.getFlattenResultIndexMapping()
+                && existingConfig.getCustomResultIndexOrAlias() != null) {
+                confirmBatchRunningListener = ActionListener
+                    .wrap(
+                        r -> getFlattenResultAliasIndex(existingConfig, listener, id, indexingDryRun),
+                        // can't update config if there is task running
+                        listener::onFailure
+                    );
 
+            } else {
+                confirmBatchRunningListener = ActionListener
+                    .wrap(
+                        r -> searchConfigInputIndices(id, indexingDryRun, listener),
+                        // can't update config if there is task running
+                        listener::onFailure
+                    );
+            }
             handler.confirmBatchRunning(id, batchTasks, confirmBatchRunningListener);
         } catch (Exception e) {
             String message = "Failed to parse config " + id;
             logger.error(message, e);
             listener.onFailure(new OpenSearchStatusException(message, RestStatus.INTERNAL_SERVER_ERROR));
         }
+    }
+
+    private void getFlattenResultAliasIndex(Config existingConfig, ActionListener<T> listener, String id, boolean indexingDryRun) {
+        String flattenResultIndexAlias = timeSeriesIndices
+            .getFlattenedResultIndexAlias(existingConfig.getCustomResultIndexOrAlias(), existingConfig.getId());
```

Comment:
existingConfig.getCustomResultIndexOrAlias can be an alias. Running get alias call on an alias can cause exception.

```
PUT my-data-stream

POST _aliases
{
  "actions": [
    {
      "add": {
        "index": "my-data-stream",
        "alias": "my-alias"
      }
    }
  ]
}

GET my-alias/_alis
```

Result:

```
{
  "error" : "no handler found for uri [/my-alias/_alis] and method [GET]"
}
```